### ISSUE TITLE
Intellij Idea with bsp doesn't make coursier download dependency sources. 

### ISSUE THREAD

**Issue by sequencer** [lihaoyi/mill#720]
After #664, mill is able to generate bsp library,
However, there still exist some problem:
1. if mill is provided by binary distribution, idea cannot find the source mill, which make the build.sc all red.
2. idea intellij won't download dependency source makes too hard to inspect API.
maybe @adadima and @samvel1024 would give a little help to this.

**Comment by sequencer** [comment]
Maybe this is related to #708

**Comment by sequencer** [comment]
As for question 2, I believe is a bug that `sourceJar` is same as `classJar`.

**Comment by sequencer** [comment]
fix by #721

**Comment by samvel1024** [comment]
I think I still have the `build.sc` symbols in read in IntelliJ after the fix 😕 
<img width="460" alt="Screen Shot 2019-10-25 at 13 59 37" src="https://user-images.githubusercontent.com/19408688/67569729-ee986880-f72f-11e9-969b-be4b3472d886.png">

**Comment by sequencer** [comment]
Yes, `build.sc` is another issue since `bsp: scala-sdk` doesn't include mill sources.(Only scala-library), This should be located in `.idea/modules/some_proj-root.iml`. Beside `mill`, it should also include any library which can be accessed by `mill -i`
Since mill is not managed by coursier, which is also self-installed jar or provided by Linux distribution, It's not easy to let idea get `CLASSES`, `JAVADOC`, `SOURCES`, I think the solution to this will probably be adding a button in IDEA to let IDEA ask mill bsp to download these sources via `MILL_VERSION` like it did in `sbt`.

**Comment by sequencer** [comment]
BTW, the `generated sources` also have some issue need to be resolved. you can find the bug when import `mill` project with bsp.

**Comment by lefou** [comment]
Fixed by #721.
